lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -613,6 +614,7 @@ def get_custom_builds_metadata(
     @router.post(
         "/api/histories/{history_id}/archive",
         summary="Archive a history.",
+        response_model_exclude_unset=True,
```

Comment:
I think that's a breaking API change. If I understand `response_model_exclude_unset` correctly fields that contain default values won't be serialized (so optional fields with `None` defaults), excluding things like `model_class` that we rely on. This would both break old scripts (I see we've been adding that option for a while now, since noone reported a bug maybe that's ok ...)

I would think the only case where this is valid is iff we have custom query keys. Can this be set more fine grained on only the partial models ?

Reply:
Also in this particular case what would result in a `CustomArchivedHistoryView` being returned ?

Reply:
> think that's a breaking API change. If I understand response_model_exclude_unset correctly fields that contain default values won't be serialized (so optional fields with None defaults), excluding things like model_class that we rely on. This would break old scripts (I see we've been adding that option for a while now, since noone reported a bug maybe that's ok ...)

> I would think the only case where this is valid is iff we have custom query keys. Can this be set more fine grained on only the partial models ?

I think `response_model_exclude_unset` does exactly what you say, but we are explicitly serializing all the fields (even with default values) when requesting each view or set of keys, so this seems fine. That also might explain why https://github.com/galaxyproject/galaxy/pull/18586#discussion_r1698385199 works... probably, it is always matching the Custom model, but since we are explicitly serializing every required field it "just works".

> Also in this particular case what would result in a CustomArchivedHistoryView being returned ?

## http://127.0.0.1:8081/api/histories/archived?keys=name,size&limit=1

```json
[
  {
    "export_record_data": null,
    "name": "MicroRNA Expression Profiling in Neurodevelopmental Disorders",
    "size": 3227245
  }
]
```

It works exactly as https://github.com/galaxyproject/galaxy/pull/18586#discussion_r1698385199 but we are always serializing `export_record_data` in every case. That might be undesired... since we didn't explicitly request that field in the keys... :thinking:

Reply:
> I think `response_model_exclude_unset` does exactly what you say, but we are explicitly serializing all the fields (even with default values) when requesting each view or set of keys, so this seems fine

Ok, that makes sense, thank you!